source/common/router/retry_state_impl.cc

```diff
@@ -25,11 +31,17 @@ RetryStatePtr RetryStateImpl::create(const RetryPolicy& route_policy,
                                      TimeSource& time_source, Upstream::ResourcePriority priority) {
   RetryStatePtr ret;
 
+  const bool conn_pool_new_stream_with_early_data_and_alt_svc = Runtime::runtimeFeatureEnabled(
+      "envoy.reloadable_features.conn_pool_new_stream_with_early_data_and_alt_svc");
   // We short circuit here and do not bother with an allocation if there is no chance we will retry.
   if (request_headers.EnvoyRetryOn() || request_headers.EnvoyRetryGrpcOn() ||
-      route_policy.retryOn()) {
+      route_policy.retryOn() ||
+      (conn_pool_new_stream_with_early_data_and_alt_svc &&
```

Comment:
Let's comment why we create a retry policy when the user didn't configure it.
Explain this will allow for retries in the case we sent an HTTP/3 request before handshake completion, the server asked for a retry after full handshake, and we're only doing it for requests which traditionally don't have bodies (no expensive buffering)

Maybe even break out the extra logic into a stanalone do_http3_early_data_retries and comment there?

Also I think we're at the point we should do release notes for new http3 features so we should doc this up in docs/root/version_history/current.rst